packages/flutter/lib/src/services/text_editing.dart

```diff
@@ -57,12 +57,25 @@ class TextRange {
   /// The text before this range.
   String textBefore(String text) {
     assert(isNormalized);
+    if(text.isNotEmpty && start > 0 && start < text.length ){
+      final int startCodeUnit = text.codeUnitAt(start);
```

Comment:
This seems dangerous.  What if the character is more than a single code unit?

Reply:
https://en.wikipedia.org/wiki/UTF-16    @dnfield     one character may be two code unit.